src/Sparrow/Threading/LightWeightThreadLocal.cs

```diff
@@ -184,9 +207,18 @@ public override bool Equals(object obj)
                     return false;
                 if (ReferenceEquals(this, obj))
                     return true;
+                if (obj.GetType() == typeof(TK))
+                {
+                    int hashCode = obj.GetHashCode();
+                    if (hashCode != _hashCode)
+                        return false;
+                    if (_weak.TryGetTarget(out var other) == false)
```

Comment:
Just to make sure - now we don't reference in `LightWeightThreadLocal._values` collection directly but via `WeakReferenceCompareValue` so it can return `false` more often, right? And that makes we we'll cleanup local thread resources - is it correct?

Reply:
Yes, that is correct.